src/v/cluster_link/source_topic_syncer.cc

```diff
@@ -143,123 +252,300 @@ ss::future<> source_topic_syncer::run_impl() {
+                });
+            }
+            continue;
+        }
+
+        // Now check to see if the the properties on the topic have differed
+        for (const auto& [key, val] : *configs) {
+            auto cached_config_it = mirror_topic_cache.topic_configs.find(key);
+            if (
+              cached_config_it == mirror_topic_cache.topic_configs.end()
+              || cached_config_it->second != val) {
+                vlog(
+                  logger().trace,
+                  "Topic {} property {} changed: {} -> {}",
+                  topic,
+                  key,
+                  cached_config_it->second,
+                  val);
+                enqueue_command = true;
+            }
+        }
+
+        if (enqueue_command) {
+            commands.emplace_back(model::update_mirror_topic_properties_cmd{
+              .topic = topic,
+              .partition_count = metadata_cache.partition_count,
+              .replication_factor = metadata_cache.rf,
+              .topic_configs = std::move(*configs)});
+        }
+    }
+}
+
+ss::future<>
+source_topic_syncer::submit_commands(reconciler_commands_vector commands) {
+    if (commands.empty()) {
+        co_return;
     }
 
-    vlog(logger().trace, "Adding mirror topics");
-    for (auto& c : add_mirror_topic_cmds) {
-        auto topic_name = c.topic;
-        vlog(logger().trace, "Adding mirror topic {}", topic_name);
-        auto res = co_await get_link()->add_mirror_topic(std::move(c));
+    for (auto& c : commands) {
+        auto res = co_await ss::visit(
+          std::move(c),
+          [this](model::add_mirror_topic_cmd c) {
+              return get_link()->add_mirror_topic(std::move(c));
+          },
+          [this](model::update_mirror_topic_properties_cmd c) {
+              return get_link()->update_mirror_topic_properties(std::move(c));
+          },
+          [this](model::update_mirror_topic_state_cmd c) {
+              return get_link()->update_mirror_topic_state(std::move(c));
+          });
         if (res != ::cluster::cluster_link::errc::success) {
-            vlog(logger().warn, "Failed to add mirror topic {}", res);
-            continue;
+            vlog(
+              logger().error,
+              "Failed to process mirror topic command: {}",
+              res);
+        } else {
+            vlog(logger().trace, "Successfully processed mirror topic command");
         }
-        vlog(logger().debug, "Successfully added mirror topic {}", topic_name);
     }
+}
 
-    if (get_state() != model::task_state::active) {
-        std::ignore = change_state(
-          model::task_state::active, "Auto topic sensor task completed");
+source_topic_syncer::candidate_update_map
+source_topic_syncer::find_candidate_topics_for_update(
+  kafka::client::cluster& cluster) {
+    // All mirror topics this link is responsible for
+    auto mirror_topics = get_link()->get_mirror_topics_for_link();
```

Comment:
this doesn't require a copy

Reply:
^^ made this change, but lmk if you want to make a case for copying out of a reference here.